File: src/helpers/getWinner.js
import Swal from "sweetalert2";

export const getWinner = (rounds, arrayRounds) => {

    if(rounds !== arrayRounds.length/2){
        Swal.fire({ icon: 'error', title: 'Oops...', text: "The number of rounds ins't correct", buttonsStyling: false, background: '#282c34' });
        return;
    }else if(rounds < 0 || rounds > 10000){
        Swal.fire({ icon: 'error', title: 'Oops...', text: "The number of round needs to be between 0 and 10000", buttonsStyling: false, background: '#282c34' });
        return;
    }

    const playerOne = [];
    const playerTwo = [];
    let winner = 0;
    let actualValuePlayerOne = 0;
    let arrayDiff = [];
    let diff = 0;

    arrayRounds.map( (value, i) => {
        if(i%2 === 0){
            playerOne.push(value);
            actualValuePlayerOne = value;
        }else {
            playerTwo.push(value);
            arrayDiff.push(Math.abs(actualValuePlayerOne - value));
        }
    })

    diff = Math.max(...arrayDiff);

    for(let i=0; i<=playerOne.length - 1; i++){
        if(Math.abs(playerOne[i] - playerTwo[i]) === diff){
            (playerOne[i] > playerTwo[i]) ? winner = 1 : winner = 2;
        }
    }

    return winner +' '+ diff;
    
}

File: src/helpers/getWinner.py
#NOTA:
#Aqui dejo el código de Python comentado del segundo ejercicio, el cual lo pueden comparar con la función de JS

# import pandas as pd

# def getWinner(rounds, array):
#     if rounds != len(array)/2:
#         print("The number of rounds isn't correct")
#         return
#     elif rounds < 0 | rounds > 10000:
#         print("The number of round needs to be between 0 and 10000")
#         return  
#     array_p1 = []
#     array_p2 = []
#     for i in array:
#         if i % 2 == 0:
#             array_p1.append(i)
#         else:
#             array_p2.append(i)
#     df = pd.DataFrame({'player1': array_p1, 'player2': array_p2}, columns=['player1', 'player2'])
#     df['diferencia'] = df['player1'] - df['player2']
#     df['abs_diff'] = abs(df['diferencia'])
#     max_diff = max(df['abs_diff'])
#     df_final = df[df['abs_diff'] == max_diff]
#     list_final = df_final.iloc[0]
#     test = int(list_final[2])
#     winner = 0
#     if test > 0:
#         winner = 1
#     else: 
#         winner = 2
#     print(winner, list_final[3])

# getWinner(5,[140, 82, 89, 134, 90, 110, 112, 106, 88, 90])